be/src/runtime/routine_load/data_consumer_group.cpp

```diff
@@ -248,13 +248,15 @@ Status KafkaDataConsumerGroup::start_all(StreamLoadContext* ctx) {
                         if (err) {
                             auto err_msg = strings::Substitute("serdes deserialize avro failed: $0", errstr);
                             LOG(ERROR) << err_msg;
+                            delete msg;
```

Comment:
How about using a smart pointer to resolve this?

Reply:
updated